draft-ietf-webtrans-http2.md

```diff
@@ -261,13 +261,67 @@ too many sessions.
 
 This document defines a WT_MAX_STREAMS capsule ({{WT_MAX_STREAMS}}) that allows
 each endpoint to limit the number of streams its peer is permitted to open as
-part of a WebTransport session. There is a separate limit for bidirectional
-streams and for unidirectional streams. Note that, unlike WebTransport over
+part of a WebTransport session.  There is a separate limit for bidirectional
+streams and for unidirectional streams.  Note that, unlike WebTransport over
 HTTP/3 {{WEBTRANSPORT-H3}}, because the entire WebTransport session is
 contained within HTTP/2 DATA frames on a single HTTP/2 stream, this limit is
 the only mechanism for an endpoint to limit the number of WebTransport streams
 that its peer can open on a session.
 
+### Initial Flow Control Limits {#flow-control-initial}
+
+To allow stream data to be exchanged in the same flight as the extended CONNECT
+request that establishes a WebTransport session, initial flow control limits
+can be exchanged via HTTP/2 SETTINGS ({{flow-control-settings}}).  Initial
+values for the flow control limits can also be exchanged via the
+`WebTransport-Init` header field on the extended CONNECT request
+({{flow-control-header}}).
+
+The limits communicated via HTTP/2 SETTINGS apply to all WebTransport sessions
+opened on that HTTP/2 connection.  Limits communicated via the
+`WebTransport-Init` header field apply only to the WebTransport session
+established by the extended CONNECT request carrying that field.
+
+If both the SETTINGS and the header field are present when a WebTransport
+session is established, the endpoint MUST use the greater of the two values for
+each corresponding initial flow control value.  Endpoints sending the SETTINGS
+and also including the header field SHOULD ensure that the header field values
+are greater than or equal to the values provided in the SETTINGS.
+
+#### Flow Control SETTINGS {#flow-control-settings}
+
+Initial flow control limits can be exchanged via HTTP/2 SETTINGS
+({{h2-settings}}) by providing non-zero values for
+
+* WT_MAX_DATA via SETTINGS_WEBTRANSPORT_INITIAL_MAX_DATA
+* WT_MAX_STREAM_DATA via SETTINGS_WEBTRANSPORT_INITIAL_MAX_STREAM_DATA_UNI and
+  SETTINGS_WEBTRANSPORT_INITIAL_MAX_STREAM_DATA_BIDI
+* WT_MAX_STREAMS via SETTINGS_WEBTRANSPORT_INITIAL_MAX_STREAMS_UNI and
+  SETTINGS_WEBTRANSPORT_INITIAL_MAX_STREAMS_BIDI
```

Comment:
Do you want to add forward references?

Reply:
Great idea, added in https://github.com/ietf-wg-webtrans/draft-ietf-webtrans-http2/pull/78/commits/13f622ff982d261277d9461bfcf03a8ab93a7617